pkg/protocol/engine/attestation/slotattestation/tracker.go

```diff
@@ -0,0 +1,88 @@
+package slotattestation
+
+import (
+	"github.com/iotaledger/hive.go/core/memstorage"
+	"github.com/iotaledger/hive.go/ds/shrinkingmap"
+	"github.com/iotaledger/hive.go/lo"
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+type Tracker struct {
+	latestAttestations  *shrinkingmap.ShrinkingMap[iotago.AccountID, *iotago.Attestation]
+	attestationsPerSlot *memstorage.IndexedStorage[iotago.SlotIndex, iotago.AccountID, *iotago.Attestation]
+
+	cutoffIndexCallback func() iotago.SlotIndex
+}
+
+func NewTracker(cutoffIndexCallback func() iotago.SlotIndex) *Tracker {
+	return &Tracker{
+		latestAttestations:  shrinkingmap.New[iotago.AccountID, *iotago.Attestation](),
+		attestationsPerSlot: memstorage.NewIndexedStorage[iotago.SlotIndex, iotago.AccountID, *iotago.Attestation](),
+
+		cutoffIndexCallback: cutoffIndexCallback,
+	}
+}
+
+func (s *Tracker) TrackAttestation(attestation *iotago.Attestation) {
+	var previousAttestationIndex iotago.SlotIndex
+	updated := true
+
+	updatedAttestation := s.latestAttestations.Compute(attestation.IssuerID, func(currentValue *iotago.Attestation, exists bool) *iotago.Attestation {
```

Comment:
Since we need more information within the Tracker I can't easily see how. Also the logic is very simple and just few lines so I don't see a big problem with it